pinot-common/src/main/java/org/apache/pinot/common/request/v2/ServerQuery.java

```diff
@@ -0,0 +1,181 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.common.request.v2;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.apache.pinot.common.request.BrokerRequest;
+
+
+public class ServerQuery {
```

Comment:
Why call this ServerQuery?

Reply:
This object can have some Server specific optimizations or pre-computed result, which might not apply to Broker. For example, we don't need to compile the filter for broker. We can add a separate BrokerQuery which only contain the information that Broker needs.
Expression, Function are common and shared for both Broker and Server.

Reply:
@kishoreg After reconsideration, changed it to QueryRequest and use it on both Broker and Server side. We can use a wrapper class on server side (E.g. ServerQueryRequest) to store the server specific helper variables.